src/beliefpropagation/beliefpropagation_schedule.jl

```diff
@@ -0,0 +1,51 @@
+abstract type EdgeSequenceAlgorithm end
+
+struct ForestCover <: EdgeSequenceAlgorithm end
+struct Parallel <: EdgeSequenceAlgorithm end
```

Comment:
Can we use the `Algorithm` type from `NDTensors` instead?

Basically you can just replace your usage of `::Parallel` with `::Algorithm"parallel"` when dispatching on `edge_sequence`, and then use `Algorithm"parallel"()` or `Algorithm("parallel")` for instances.